pkg/dfns-api-client/auth_transport.go

```diff
@@ -0,0 +1,276 @@
+package dfns_api_client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"slices"
+	"strconv"
+
+	"github/dfns/dfns-sdk-go/pkg/credentials"
+)
+
+const (
+	UserActionHeader    = "x-dfns-useraction"
```

Comment:
nit: maybe add a comment just to say that this one is meant for client-side only (not passed in request) if someone looks at your code